bundles/framework/myplaces3/view/MainView.js

```diff
@@ -163,60 +164,30 @@ Oskari.clazz.define('Oskari.mapframework.bundle.myplaces3.view.MainView',
          */
         showPlaceForm: function (location, place) {
             const me = this;
-            let layerId;
             const sandbox = this.instance.sandbox;
+            const categoryHandler = this.instance.getCategoryHandler();
+            const categories = categoryHandler.getAllCategories();
+
             sandbox.postRequestByName('DisableMapKeyboardMovementRequest');
+
             this.form = Oskari.clazz.create(
                 'Oskari.mapframework.bundle.myplaces3.view.PlaceForm',
                 this.instance,
-                this.options
+                this.options,
+                categories
             );
-            const categoryHandler = this.instance.getCategoryHandler();
+
+            this.form.setDrawing(this.drawing);
+
             if (place) {
-                layerId = categoryHandler.getMapLayerId(place.getCategoryId());
                 this.isEditPlace = true;
             // set measurement result from drawing
             } else {
                 this._updateMeasurementResult(this.drawingData);
             }
-            var categories = categoryHandler.getAllCategories();
-            var formEl = me.form.getForm(categories, place);
-            var actions = [
-                {
-                    name: me.loc('placeform.category.newLayer'),
-                    type: 'link',
-                    group: 0,
-                    selector: '#newLayerForm > label',
-                    action: function () {
-                        me.form.createCategoryForm();
-                    }
-                }, {
-                    name: me.loc('buttons.cancel'),
-                    type: 'button',
-                    group: 1,
-                    action: function () {
-                        me.cleanupPopup();
-                    }
-                }, {
-                    name: me.loc('buttons.save'),
-                    type: 'button',
-                    group: 1,
-                    action: function () {
-                        me._saveForm();
-                    }
-                }
-            ];
-            const content = [{
-                html: formEl,
-                actions,
-                layerId
-            }];
 
-            var options = {
-                hidePrevious: true
-            };
-            var request = Oskari.requestBuilder('InfoBox.ShowInfoBoxRequest')(this.popupId, me.loc('placeform.title'), content, location, options);
-            sandbox.request(me.getName(), request);
+            me.form.getForm(categories, place); // Get form
```

Comment:
The getForm() is a misleading name since the return value is not used. Does it show the form?

Reply:
Changed `getForm` -> `showForm`